packages/core/src/provider.ts

```diff
@@ -339,14 +339,15 @@ export async function getBalance(
 ): Promise<Balances | null> {
   // chain we don't recognize and don't have a rpcUrl for requests
   if (!chain) return null
-
-  const provider = getProvider(chain)
+  
+  const { wallets } = state.get()
 
   try {
-    const balanceWei = await provider.getBalance(address)
-
-    return balanceWei
-      ? { [chain.token || 'eth']: utils.formatEther(balanceWei) }
+    const wallet = wallets.find(wallet => wallet.label)
+    const provider = wallet.provider
+    const balanceHex = await provider.request({ method: 'eth_getBalance', params:[address,'latest'] });
+    return balanceHex
+      ? { [chain.token || 'eth']: shortenHexToString(balanceHex) }
```

Comment:
I don't think we want to shorten the balance string here and if we were to do that we should probably round the number. The balance should just return the full balance as a decimal string. Then wherever the balance gets rendered, it can be decided how many decimal places should be rendered.

Reply:
Hmmn. Makes sense. So I reverted it back to the original ethers BigNumber "BalanceWei' version but using 'bignumber.js' instead.